shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/repository/AuthRepositoryImpl.kt

```diff
@@ -0,0 +1,37 @@
+package co.nimblehq.mark.kmmic.data.repository
+
+import co.nimblehq.mark.kmmic.data.service.auth.AuthService
+import co.nimblehq.mark.kmmic.data.service.auth.model.AuthLoginParams
+import co.nimblehq.mark.kmmic.domain.model.AuthToken
+import co.nimblehq.mark.kmmic.domain.repository.AuthRepository
+import com.russhwolf.settings.ExperimentalSettingsApi
+import com.russhwolf.settings.Settings
+import com.russhwolf.settings.serialization.decodeValueOrNull
+import com.russhwolf.settings.serialization.encodeValue
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlinx.serialization.ExperimentalSerializationApi
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+private const val KEY_ACCESS_TOKEN = "access_token"
+
+@OptIn(ExperimentalSerializationApi::class, ExperimentalSettingsApi::class)
+internal class AuthRepositoryImpl: AuthRepository, KoinComponent {
+    private val authService: AuthService by inject()
```

Comment:
Just curious, how can we mock these properties in unit tests? Or should we use constructor injection instead?

Reply:
@AVI5HEK you can check this [file](https://github.com/markgravity/kmm-ic/blob/3225037d11388c90ad505fdf1de85c0ac1d7d6f2/shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/domain/usecase/GetCurrentUserUseCaseTest.kt) to see how I write test 🙏

Reply:
@AVI5HEK but it will make it harder to call it in the native app cause we need to use the Koin instance to create the use case 🙏

Reply:
Can't we use this in `DI.kt`?
<img width="665" alt="image" src="https://user-images.githubusercontent.com/8093908/233971014-0774bb22-d84f-4f3e-89af-efac3da7bc9b.png">

And here's the constructor:
<img width="629" alt="image" src="https://user-images.githubusercontent.com/8093908/233971181-4c9f0201-09eb-43c0-8a2e-e2a4a9b0140d.png">

Reply:
@AVI5HEK We need some extra code to create an instance on our side, making it more dependent on the Koin lib, also what I'm trying to archive
- Make it like a normal framework, so that we can create instances in the normal way:
```
useCase = KoinInstance.koin["logInUseCase"]
// vs with the normal way
useCase = LoginUseCaseImpl()
```
- We are using the same approach to inject the instance to write the test
```
Container.shared.loginUseCase.register { loginUseCaseMock }
viewModel = LoginViewModel()
```

Reply:
I'm okay with whichever is more convenient. 👌🏽